### PR TITLE
REF: Share NumericArray/NumericDtype methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — COMMENTED** [review]
Off-topic question: I am interested in exploring the feasibility of adding an Arrow-back NumericArray. It doesn't appear that I can just subclass this since NumericArray and it's subclass are heavily numpy based.

Any recommendations on where to start? Would I need to essentially create `MaskedArrowArray` and go from there?

File: pandas/core/arrays/numeric.py [hunk 1]
```diff
@@ -41,6 +44,22 @@
 
 class NumericDtype(BaseMaskedDtype):
     _default_np_dtype: np.dtype
+    _checker: Callable[[Any], bool]  # is_foo_dtype
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/numeric.py:47]: sure.  OK for follow-up?

**Comment by jbrockmendel** [comment]
> Any recommendations on where to start? Would I need to essentially create MaskedArrowArray and go from there?

I've been thinking we should refactor as much of ArrowStringArray as possible out into a general arrow-backed EA.  Not sure if that could be a fully-formed EA or if it would need to be a mixin.  (also in the tests there's an ArrowExtensionArray but I think its half-baked)

FWIW my current thought with MaskedArray is to try to get as much of NumericArray and BooleanArray pushed down into MaskedArray as possible so it can just become a wrapper around an arbitrary non-masked array (including e.g. other EAs)

File: pandas/core/arrays/numeric.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/arrays/numeric.py:47]: Yup sounds good

**Review by mroeschke — APPROVED** [review]
Thanks for the gameplan. LGTM.

**Comment by jreback** [comment]
luv it

**Pull request merged**